### ISSUE TITLE
Delete Config File?

### ISSUE THREAD

**Issue by trisys3** [sdepold/gulp-busterjs#5]
I was recently going through this task's code and, while I'm not a master of Node by any means, I noticed it deletes a config file if it finds any in the current folder before it generates a new one. Now, my first question is: do we even need to delete this file? Gulp operates on streams, which means you don't need to do operations on actual files, but rather on the virtual `vinyl-fs` provided. Therefore, unless we were explicitly creating a new config file and adding it in (which we aren't), we shouldn't need to delete the old one.

We could even delete the config file in the `vinyl-fs` and generate a new one, but not "commit" this operation to the filesystem. We shouldn't even need to do this, however, because we could just generate a new one in vinyl and not even touch the filesystem.

A better way seems to be to take the options in the repo's current config file into the stream, then merge in some options we provide, deleting any that conflict. This way, we take into account the user's config file, not assuming anything, then merge the options the user provided in the task, then provide defaults if any options are still missing

I am not worried about my buster file being deleted, by the way, as it is in a subfolder, and I have my repo on git anyway. I am worried about any task that deletes my files without me telling it to, however, as, even if I know what it deletes now, if it updates it could pull a fast one before I realize what it's done. I am not suggesting this is what I think of this task, merely suggesting that this is what other users may think.

**Comment by sdepold** [comment]
Hi there,

I'm no master of Gulp by any means :) I didn't even know about that `vinyl-fs` thing. Can I spawn a process which runs on `vinyl-fs`? The buster config file is deleted and generated by this gulp tasks accordingly to the config you pass in.

**Comment by trisys3** [comment]
Yeah, I noticed I got the order of operations wrong. I said the plugin deletes the file first, then generates a new one, while it's actually the opposite. That's why I closed this issue, so I would hopefully not make more of a fool out of myself as I figured out if my fears were in fact warranted.

As to your question about `vinyl-fs`, as I said, I am not a Node master. I knew enough to know that it exists, but have not dug through its code yet, nor that of other `gulp` plugins. I think I'm still reeling from the times when I tried that for `grunt`, and got really confused as a result. Trouble is, sometimes it seems you have to know quite a bit about the underlying workings of `gulp` to use it even as a developer, which was far from the case for `grunt`. I will try to look into this for you.

**Comment by trisys3** [comment]
Taking a quick look at some `gulp` plugins, it looks like many of them are using `through2`, which seems like it handles streams for you, just like `vinyl-fs`. They seem pretty similar, in fact. One plugin that I saw, `gulp-connect`, uses `fs` to read some files. This is all in contrast to this module, which actually writes to files with `fs`.

**Comment by sdepold** [comment]
From what I understood after taking a quick glance at vinyl-fs it would write to the hard drive as well. Also I'm not sure if I could really spawn a buster command while reading from an in-memory hard drive (which would probably be possible). Ideally I wouldn't even need to spawn a process for buster and just call the respective piece of code in the gulp task itself, but buster makes it overly hard to actually trigger the test run while bypassing the executable. Because of that I would keep the current flow for now.

**Comment by trisys3** [comment]
According to `vinyl-fs`'s README, you have to trigger no-write mode manually and anyway, as I stated before, it looks like most `gulp` plugins are using `through2`, not `vinyl-fs`. `vinyl-fs` is what `gulp` uses internally, but I suppose plugins don't have to. That bit about not being able to bypass the executable would still be an issue, though.

Anyway, if I find time (and learn more streams) I will see if I can submit a pull-request for you.

**Comment by sdepold** [comment]
That would be great:)
